### PR TITLE
feat: add `default_values` to interface

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/symbol_cache.jl [hunk 1]
```diff
@@ -75,6 +75,7 @@ all_variable_symbols(sc::SymbolCache) = variable_symbols(sc)
 function all_symbols(sc::SymbolCache)
     vcat(variable_symbols(sc), parameter_symbols(sc), independent_variable_symbols(sc))
 end
+default_values(::SymbolCache) = Dict()
```
Comment (SebastianM-C) [inline src/symbol_cache.jl:78]: Does this create a new dictionary every time we need defaults?

**Review by SebastianM-C — COMMENTED** [review]
Does the spec says what should be used as keys or is it up to the package that implements this? I was thinking that if we use something that standardized, then the defaults are compatible between packages implementing the interface.

Also, are the `default_values` something that's supposed to be immutable or not?

**Review by SebastianM-C — APPROVED** [review]
The mutability of the defaults dictionary is needed for updating the default values?

**Comment by AayushSabharwal** [comment]
Right now it's somewhat arbitrary. I don't think we'll need it to be mutable in remake or anywhere else, but you mentioned that we should standardize it and mutability offers us more flexibility in the future.

**Pull request merged**